tests_e2e/tests/agent_ext_workflow/extension_workflow.py

```diff
@@ -0,0 +1,358 @@
+            status_message = instance_view.statuses[0].message
+            log.info("Status message: %s" % status_message)
+
+            with soft_assertions():
+                expected_ext_version = "%s-%s" % (self.name, self.version)
+                assert_that(expected_ext_version in status_message).described_as(
+                    f"Specific extension version name should be in the InstanceView message ({expected_ext_version})").is_true()
+
+                expected_count = "%s: %s" % (self.COUNT_KEY_NAME, self.enable_count)
+                assert_that(expected_count in status_message).described_as(
+                    f"Expected count should be in the InstanceView message ({expected_count})").is_true()
+
+                if self.data is not None:
+                    expected_data = "{0}: {1}".format(self.DATA_KEY_NAME, self.data)
+                    assert_that(expected_data in status_message).described_as(
+                        f"Expected data should be in the InstanceView message ({expected_data})").is_true()
+
+        def execute_assertion_script(self, file_name, args):
+            log.info("")
+            log.info("Running {0} remotely with arguments {1}".format(file_name, args))
+            result = self.ssh_client.run_command(f"{file_name} {args}", use_sudo=True)
+            log.info(result)
+            log.info("Assertion completed successfully")
+
+        def assert_scenario(self, file_name: str, command_args: str, assert_status: bool = False, restart_agent: list = None, data: str = None):
+            # Assert the status in the instance view
+            if assert_status:
+                self.assert_instance_view(data=data)
+
+            # Remotely execute the assertion script
+            self.execute_assertion_script(file_name, command_args)
+
+            # Restart the agent and test the status again if enabled (by checking the operations.log file in the VM)
+            # Restarting agent should just run enable again and rerun the same settings
+            if restart_agent is not None:
+                log.info("")
+                log.info("Restarting the agent...")
+                output = self.ssh_client.run_command("agent-service restart", use_sudo=True)
+                log.info("Restart completed:\n%s", output)
+
+                for args in restart_agent:
+                    self.execute_assertion_script('assert-operation-sequence.py', args)
+
+                if assert_status:
+                    self.assert_instance_view()
+
+        def update_ext_version(self, extension: VirtualMachineExtensionClient, version: str):
+            self.extension = extension
+            self.version = version
+
+    def run(self):
+        is_arm64: bool = self._ssh_client.get_architecture() == "aarch64"
+
+        if is_arm64:
+            log.info("Skipping test case for %s, since it has not been published on ARM64", VmExtensionIds.GuestAgentDcrTestExtension)
```

Comment:
let's publish an ARM64 package

Reply:
Holding off on this. We're thinking about writing the test extension in Rust and publishing to arm64/all regions with the rust extension